### ISSUE TITLE
message signing with eth.sign() broken

### ISSUE THREAD

**Issue by brenzi** [paritytech/parity#5568]
I'm trying to sign a text message with the private key of my ethereum wallet, but the function seems to be broken.

reproduce (web3 console):
```
> address='0x00106fE095f556e6E4d18BE78CfC263D27c1D2C2'
'0x00106fE095f556e6E4d18BE78CfC263D27c1D2C2'
> plain="hallo"
'hallo'
> msg=web3.sha3(plain)
'0xd123f0a85b26264de171387e5503b75ee1f33737c297fdba8c43a3191ff2f8d0'
> web3.eth.sign(address,msg)
```
now parity trusted signer asks me for the passphrase. Then I get:
```
'0x1bb640d5f69ba246fa909f06d5ac1b343b4c241d2485daab1d4aa53feb950eddbe0fe6fa42c6f0dd3a0b5d70153089268aa58b1c64f24d7626a14d34fd1e46e4d2'
```
now, this signature doesn't verify on https://etherchain.org/verify/signature
nor manually (using ethereumjs-utils):
```
sign='0x1bb640d5f69ba246fa909f06d5ac1b343b4c241d2485daab1d4aa53feb950eddbe0fe6fa42c6f0dd3a0b5d70153089268aa58b1c64f24d7626a14d34fd1e46e4d2'
r = utils.toBuffer(sgn.slice(0,66))
s = utils.toBuffer('0x' + sgn.slice(66,130))
v = utils.toBuffer('0x' + sgn.slice(130,132))
m = utils.toBuffer(msg)
pub = utils.ecrecover(m, v, r, s)
```
ecrecover fails:
```
Error: Invalid signature v value
    at Object.exports.ecrecover (/home/brenzi/node_modules/ethereumjs-util/index.js:362:11)
    at repl:1:13
    at sigintHandlersWrap (vm.js:22:35)
    at sigintHandlersWrap (vm.js:96:12)
    at ContextifyScript.Script.runInThisContext (vm.js:21:12)
    at REPLServer.defaultEval (repl.js:313:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:513:10)
    at emitOne (events.js:101:20)
```
This seems to work for others (geth users?):
https://www.reddit.com/r/ethereum/comments/4yamv4/signing_a_simple_message/
that's why I wonder if parity users can sign and verify messages

thanks for your help

**Comment by 5chdn** [comment]
This might be the issue here #5490

**Comment by 5chdn** [comment]
I merged your comment to #5490 if you don't mind subscribe it.
